test/end-to-end/utils/coverage.js

```diff
@@ -7,18 +7,35 @@
 // collects coverage from the tested page,
 // calls nightmare.end() and calls done()
 // executed via eval() as a last step of every test!
+if (pageConfig.root.includes('9090')) {
+  nightmare
+    .exitIFrame()
+    .evaluate(() => document.getElementsByClassName('container-frame')[0].contentWindow.__coverage__)
```

Comment:
IMO the check whether or not we are running inside Cockpit should be done inside the evaluate method and return the appropriate variable for coverage.

Questions:

1) From inside the iframe can we access the URL of the browser and check if we are running in Cockpit?
2) Why do we have to exit the iframe and then access its `.__coverage__` object from the parent? Is there any way to access this object even if we are inside the iframe (even if you have to access the parent from inside the iframe) ?

Reply:
To exit iframe first is window.__coverage__ cannot be fetched inside iframe. So I have to exit, then get coverage from global.

Reply:
Can you get access to the parent page without exiting the iframe?

Reply:
Accessing parent page works, but still cannot get __coverage__ value.

Reply:
if you can access the parent object can't you do something like `document.parent.getElementsByClassName('container-frame')[0].contentWindow.__coverage__` ? What does "Accessing parent page works, but still cannot get coverage value." mean? If you can access the parent object what is preventing you from accessing the coverage object as well ?

Reply:
That'll report `document.parent is undefined` error.

Reply:
OK, don't take what I write verbatim. Show me how do you access the parent page from within the iframe ?